cmd/kube-apiserver/app/server.go

```diff
@@ -418,6 +419,16 @@ func buildGenericConfig(
 		return
 	}
 
+	if genericConfig.EgressSelector != nil {
```

Comment:
Is EgressSelector by default nil?
Do we have test coverage in case where EgressSelector is not nil, and the cluster network isn't directly accessible by the master?

Reply:
Yes it is nil by default. Hmm not aware of it if any, @cheftako maybe?

Reply:
Yes EgressSelector is by default nil. It's only initialized if we pass in a flag to the api server (off by default) https://github.com/kubernetes/kubernetes/blob/c7869131dd2a7272bd329a4cd9c3abb7f3805217/cluster/gce/gci/configure-kubeapiserver.sh#L331

We don't have any tests for that scenario...although since the tunnels are created via a request from cluster to master,    that scenario implies that the konnectivity server started and no agents registered with it. This technically happens every time a cluster starts up since the KAS & konnectivity server on the master will be ready much quicker than the the konnectivity agent pod on the node.